test/test-remote/test_dd.ts

```diff
@@ -115,5 +97,81 @@ suite("DD API test suite", function () {
       https: { rejectUnauthorized: false }
     });
     logHTTPResponse(response);
+
+    // now query cortex
+    const searchStart = now.minusMinutes(45);
+    const searchEnd = now.plusMinutes(10);
+
+    const queryParams = {
+      query: `${metricnameSanitized}{job="ddagent"}`,
+      start: searchStart.toEpochSecond().toString(),
+      end: searchEnd.toEpochSecond().toString(),
+      step: "60s"
+    };
+
+    const resultArray = await waitForCortexQueryResult(
+      TENANT_DEFAULT_CORTEX_API_BASE_URL,
+      queryParams
+    );
+
+    log.info("resultArray: %s", JSON.stringify(resultArray, null, 2));
+
+    // When the `points` array in the submit request is ascending in time
+    // (newer samples towards the end of the array) then the result reflects
+    // all values:
+
+    // "values": [
+    //     [
+    //       1611598200,
+    //       "0"
+    //     ],
+    //     [
+    //       1611598260,
+    //       "0"
+    //     ],
+    //     [
+    //       1611598320,
+    //       "1"
+    //     ],
+    //     [
+    //       1611598380,
+    //       "1"
+    //     ],
+    //     [
+    //       1611598440,
+    //       "2"
+    //     ]
+    //   ]
+
+    // When the `points` array in the submit request is descending in time
+    // (newer samples towards the beginning of the array, which is what the DD
+    // agent seems to send) then the result reflects only the newest sample:
+    // all values:
+
+    // "values": [
+    //     [
+    //       1611599040,
+    //       "2"
+    //     ]
+
+    // The latter case seems to imply information loss with the current
+    // DD->Cortex translation logic (which simply iterates over DD
+    // agent-provided samples in the order as given in the JSON doc).
+    // Interestingly.
+
```

Comment:
`// Samples within a time series must be ordered by time.`, from https://godoc.org/github.com/prometheus/prometheus/prompb -- but documented for type QueryResult. Mhm.